api/spec/requests/spree/api/v2/storefront/cart_spec.rb

```diff
@@ -410,7 +410,65 @@
     context 'without coupon code' do
       context 'does not apply the coupon code' do
         before do
-          patch '/api/v2/storefront/cart/apply_coupon_code', params: { user: user, coupon_code: '' }, headers: headers
+          patch '/api/v2/storefront/cart/apply_coupon_code', params: { coupon_code: '' }, headers: headers
+        end
+
+        it 'returns 422 status with an error' do
+          expect(response.status).to eq(422)
+
+          expect(json_response[:error]).to eq("The coupon code you entered doesn't exist. Please try again.")
```

Comment:
@damianlegawiec maybe better to take expected value text from I18n but maybe this way also good.

Reply:
@gingray we usually don't use I18n in tests to not be surprised by some side-effects